### PR TITLE
refactor(Core/BattlegroundSA): Convert to std::chrono

### PR DESCRIPTION
**Pull request by Kitzunu**
## Changes Proposed:

* Use `std::chrono`

## Issues Addressed:

- ref https://github.com/azerothcore/azerothcore-wotlk/issues/6490

## Tests Performed:

- Builds

## How to Test the Changes:

1. Check that all the timers in SA works the same as before

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Kitzunu** [comment]
@azerothcore/testers

File: src/server/game/Battlegrounds/Zones/BattlegroundSA.h [hunk 1]
```diff
@@ -52,12 +52,9 @@ enum BG_SA_GateState
...
 
-enum BG_SA_Timers
-{
-    BG_SA_BOAT_START    =  60 * IN_MILLISECONDS,
-    BG_SA_WARMUPLENGTH  = 120 * IN_MILLISECONDS,
-    BG_SA_ROUNDLENGTH   = 600 * IN_MILLISECONDS
-};
+static constexpr Milliseconds BG_SA_BOAT_START    = 1min;
+static constexpr Milliseconds BG_SA_WARMUPLENGTH  = 2min;
+static constexpr Milliseconds BG_SA_ROUNDLENGTH   = 10min;
```
Comment (Winfidonarleyan) [inline src/server/game/Battlegrounds/Zones/BattlegroundSA.h:57]: in `.cpp` and no need use `static`

**Comment by claudiodfc** [comment]
Tested.
- Strand of the Ancients starts in 2 minutes.
- The boat starts moving at 1 minute left to begin.
- There's a countdown message until it begins.
- Announcer saying the battle begun after 2 minutes.
- The timer starts at 10 minutes

File: src/server/game/Battlegrounds/Zones/BattlegroundSA.h (hunk 1 above)
Comment (Nyeriah) [reply src/server/game/Battlegrounds/Zones/BattlegroundSA.h:57]: @Kitzunu

**Pull request merged**